src/Gateway.sol

```diff
@@ -593,7 +593,8 @@ contract Gateway is IGateway, IExecutor, IUpgradable, GatewayEIP712 {
      * Deposit funds to the gateway contract
      * IMPORTANT: this function must be called only by the administrator!!!!
      */
-    function deposit() external payable {}
+    receive() external payable {}
+    fallback() external payable {}
```

Comment:
Just the `receive` method is necessary, also do this check:
```
require(msg.data.length == 0);
```

Otherwise it can open space to some vulnerabilities regarding third part contracts.